internal/engine/pod.go

```diff
@@ -114,35 +117,30 @@ func matchPodChangeToManifest(state *store.EngineState, action k8swatch.PodChang
 func maybeTrackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*store.Pod, bool) {
 	pod := action.Pod
 	podID := k8s.PodIDFromPod(pod)
-	startedAt := pod.CreationTimestamp.Time
-	status := k8swatch.PodStatusToString(*pod)
-	ns := k8s.NamespaceFromPod(pod)
-	hasSynclet := sidecar.PodSpecContainsSynclet(pod.Spec)
 	runtime := ms.GetOrCreateK8sRuntimeState()
 	isCurrentDeploy := runtime.HasOKPodTemplateSpecHash(pod) // is pod from the most recent Tilt deploy?
 
+	// Only attach a new pod to the runtime state if it's from the current deploy;
+	// if it's from an old deploy/an old Tilt run, we don't want to be checking it
+	// for status etc.
+	if !isCurrentDeploy {
+		return runtime.Pods[podID], false
+	}
+
 	// Case 1: We haven't seen pods for this ancestor yet.
 	ancestorUID := action.AncestorUID
 	isAncestorMatch := ancestorUID != ""
 	if runtime.PodAncestorUID == "" ||
 		(isAncestorMatch && runtime.PodAncestorUID != ancestorUID) {
 		podInfo := &store.Pod{
-			PodID:      podID,
-			StartedAt:  startedAt,
-			Status:     status,
-			Namespace:  ns,
-			HasSynclet: hasSynclet,
-			SpanID:     runtimelog.SpanIDForPod(podID),
-		}
-		if isCurrentDeploy {
-			// Only attach a new pod to the runtime state if it's from the current deploy;
-			// if it's from an old deploy/an old Tilt run, we don't want to be checking it
-			// for status etc.
-			runtime.PodAncestorUID = ancestorUID
-			runtime.Pods = make(map[k8s.PodID]*store.Pod)
-			runtime.Pods[podID] = podInfo
-			ms.RuntimeState = runtime
+			PodID: podID,
 		}
+
+		runtime.PodAncestorUID = ancestorUID
+		runtime.Pods = make(map[k8s.PodID]*store.Pod)
```

Comment:
any reason not to initialize this map in `GetOrCreateK8sRuntimeState`?

Reply:
we're clearing it, not initializing it. will add a comment on that too